src/main/scala/chisel3/simulator/package.scala

```diff
@@ -117,21 +120,26 @@ package object simulator {
             if (filename.startsWith(workspace.supportArtifactsPath)) {
               filename = filename.substring(workspace.supportArtifactsPath.length + 1)
             }
-            java.nio.file.Files.move(
-              java.nio.file.Paths.get(s"${workspace.supportArtifactsPath}/$filename"),
-              java.nio.file.Paths.get(s"${workspace.primarySourcesPath}/$filename")
-            )
+            val src = java.nio.file.Paths.get(s"${workspace.supportArtifactsPath}/$filename").normalize()
+            if (!movedFiles.contains(src)) {
+              java.nio.file.Files.move(
+                src,
+                java.nio.file.Paths.get(s"${workspace.primarySourcesPath}/$filename")
+              )
+              movedFiles.addOne(src)
```

Comment:
Nit but typical style would just be `movedFiles += src`

Reply:
Fixed in the force push!